apps/desktop-wallet/src/modals/AddressDetailsModal.tsx

```diff
@@ -32,30 +32,33 @@ import HashEllipsed from '@/components/HashEllipsed'
 import QRCode from '@/components/QRCode'
 import TransactionList from '@/components/TransactionList'
 import AssetsTabs from '@/features/assetsLists/AssetsTabs'
-import { useAppSelector } from '@/hooks/redux'
-import CSVExportModal from '@/modals/CSVExportModal'
-import ModalPortal from '@/modals/ModalPortal'
+import { closeModal, openModal } from '@/features/modals/modalActions'
+import { ModalBaseProp } from '@/features/modals/modalTypes'
+import { useAppDispatch, useAppSelector } from '@/hooks/redux'
 import SideModal from '@/modals/SideModal'
 import AmountsOverviewPanel from '@/pages/UnlockedWallet/OverviewPage/AmountsOverviewPanel'
 import { selectAddressByHash } from '@/storage/addresses/addressesSelectors'
 import { openInWebBrowser } from '@/utils/misc'
 
-interface AddressDetailsModalProps {
+export interface AddressDetailsModalProps {
   addressHash: AddressHash
-  onClose: () => void
 }
 
-const AddressDetailsModal = ({ addressHash, onClose }: AddressDetailsModalProps) => {
+const AddressDetailsModal = memo(({ id, addressHash }: ModalBaseProp & AddressDetailsModalProps) => {
```

Comment:
> `memo` lets you skip re-rendering a component when its props are unchanged.
https://react.dev/reference/react/memo

We need to make sure that **any** modal can render on top of **any other** modal. We also need to make sure that the same modal can appear **multiple times** in the stack of modals. To achieve this, as seen later, in the `AppModals.tsx` component, we are loop over an array of opened modals to render them.

`memo` is used to wrap every (refactored) modal component. The purpose of it is to avoid re-rendering a modal that is already open **when a new open opens** (since the open arrays modal changes when a new modal opens, each modal is a new instance - we want to avoid re-rendering modal instance components when their props haven't changed).